tests/core/test_graph.py

```diff
@@ -1,54 +1,367 @@
+# pylint: disable=missing-function-docstring,missing-module-docstring
+# pylint: disable=invalid-name
+
 from __future__ import annotations
 
+import pytest
+import scipy.sparse as s
 import torch
+from torch_geometric.edge_index import EdgeIndex
+from torch_geometric.data import Data
+from torch_geometric.testing import get_random_edge_index
+
+from pathpyG import Graph, IndexMap
+
+
+def test_init():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    data = Data(edge_index=edge_index, num_nodes=100)
+    g = Graph(data)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_edge_index():
+    edge_index = EdgeIndex(get_random_edge_index(100, 100, 1000))
+    data = Data(edge_index=edge_index, num_nodes=100)
+    g = Graph(data)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_mapping():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    data = Data(edge_index=edge_index, num_nodes=100)
+    idxs = [str(i) for i in range(100)]
+    mapping = IndexMap(idxs)
+    g = Graph(data, mapping)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_init_with_list_mapping():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    data = Data(edge_index=edge_index, num_nodes=100)
+    idxs = [str(i) for i in range(100)]
+    g = Graph(data, idxs)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_from_edge_index():
+    edge_index = get_random_edge_index(100, 100, 1000)
+    g = Graph.from_edge_index(edge_index)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_from_edge_list():
+    edge_list = [
+        ("a", "b"),
+        ("b", "c"),
+        ("c", "a"),
+    ]
+    g = Graph.from_edge_list(edge_list)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_from_edge_list_undirected():
+    edge_list = [
+        ("a", "b"),
+        ("b", "a"),
+        ("b", "c"),
+        ("c", "b"),
+        ("c", "a"),
+        ("a", "c"),
+    ]
+    g = Graph.from_edge_list(edge_list, is_undirected=True)
+    assert isinstance(g, Graph)
+    assert isinstance(g.data, Data)
+    assert isinstance(g.mapping, IndexMap)
+    assert isinstance(g.edge_to_index, dict)
+
+
+def test_from_edge_list_fails():
```

Comment:
Is this a behaviour that we want or do we trust the users to only input correct things?

Reply:
This is the intended behavior, i.e. we require the user to give correct input, raising an exception if the input is wrong.